pkg/config/config_parser.go

```diff
@@ -150,22 +153,42 @@ func decodeConfig(r io.Reader) (*Config, diag.Diagnostics) {
 		return nil, diags
 	}
 
+	var diags diag.Diagnostics
 	providers := yc.Providers
-	diags := diag.Diagnostics{}
 	for _, p := range providers {
 		p.Configuration, err = yaml.Marshal(p.ConfigKeys["configuration"])
 		if err != nil {
 			diags = diags.Add(diag.FromError(err, diag.INTERNAL, diag.WithSummary("ConfigKeys marshal failed")))
 			continue
 		}
+		delete(p.ConfigKeys, "configuration")
```

Comment:
nit: document why you are deleting the key

Reply:
solved in new commit